applications/ShapeOptimizationApplication/python_scripts/response_functions/packaging_response_base.py

```diff
@@ -0,0 +1,157 @@
+
+import time as timer
+
+import KratosMultiphysics as KM
+from KratosMultiphysics import Logger
+
+from .response_function import ResponseFunctionBase
+
+
+class PackagingResponseBase(ResponseFunctionBase):
+    """
+    A base class for packaging response functions that agglomerate the nodal violations
+    into a single response function.
+    The agglomeration happens by summing up the square of each nodal violation.
+    Nodes that are feasible do NOT contribute to the response value/gradient.
+    This is why a prediction of the violation using the gradients is not possible,
+    only correction of violations (e.g. from the last step) will happen.
+
+    Derived classes need to implement the calculation of the nodal violations
+
+    Important settings:
+    infeasible_side : boolean flag that indicates if the normal of bounding instance
+        points to the infeasible side. False by default
+    """
+
+    def __init__(self, identifier, response_settings, model):
+        self.identifier = identifier
+
+        response_settings.ValidateAndAssignDefaults(self.GetDefaultSettings())
+
+        self.response_settings = response_settings
+        self.model = model
+        self.model_part_needs_to_be_imported = False
+
+        model_part_name = response_settings["model_part_name"].GetString()
+        input_type = response_settings["model_import_settings"]["input_type"].GetString()
+        if input_type == "mdpa":
+            self.model_part = self.model.CreateModelPart(model_part_name)
+            domain_size = response_settings["domain_size"].GetInt()
+            if domain_size not in [2, 3]:
+                raise Exception("PackagingResponseBase: Invalid 'domain_size': {}".format(domain_size))
+            self.model_part.ProcessInfo.SetValue(KratosMultiphysics.DOMAIN_SIZE, domain_size)
+            self.model_part_needs_to_be_imported = True
+        elif input_type == "use_input_model_part":
+            self.model_part = self.model.GetModelPart(model_part_name)
+        else:
+            raise Exception("PackagingResponseBase: '{}' model part input type not implemented.".format(input_type))
+
+        self.signed_distances = None
+        self.directions = None
+
+        self.value = None
+
+        self.gradient = {}
+
+        self.infeasible_side = self.response_settings["infeasible_side"].GetBool()
+        self.exponent = 2
+
+    @classmethod
+    def GetDefaultSettings(cls):
+        this_defaults = KM.Parameters("""{
+            "response_type"         : "plane_packaging",
+            "model_part_name"       : "UNKNOWN_NAME",
+            "domain_size"           : 3,
+            "model_import_settings" : {
+                "input_type"        : "use_input_model_part",
+                "input_filename"    : "UNKNOWN_NAME"
+            },
+            "infeasible_side"       : false
+        }""")
+        return this_defaults
+
+    def Initialize(self):
+        if self.model_part_needs_to_be_imported:
```

Comment:
```suggestion
        if self.response_settings["model_import_settings"]["input_type"].GetString() == "mdpa":
```
This way you would not need the extra member variable `self.model_part_needs_to_be_imported`